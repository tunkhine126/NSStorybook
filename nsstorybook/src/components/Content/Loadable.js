/**
 *
 * Asynchronously loads the component for Content
 *
 */
import React from 'react';
import loadable from 'utils/loadable';
import Loading from 'components/shared/Loading';

export default loadable(() => import('./index'), {
  fallback: <Loading />,
});
