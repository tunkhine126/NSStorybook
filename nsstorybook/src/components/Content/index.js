/**
 *
 * Content
 *
 */

import React from 'react';
import PropTypes from 'prop-types';

import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';

export function Content({
  subHeader,
  body,
  headerClass,
  containerClass,
  screen,
}) {
  const headerScreen = (screen && screen.header.xs) || 3;
  const bodyScreen = (screen && screen.header.xs) || 9;

  return (
    <>
      <Grid
        container
        direction="row"
        justify="flex-start"
        alignItems="flex-start"
        className={containerClass}
      >
        <Grid item xs={headerScreen}>
          <Typography variant="body1" className={headerClass} gutterBottom>
            {subHeader}
          </Typography>
        </Grid>
        <Grid item xs={bodyScreen}>
          <Grid
            container
            direction="column"
            justify="flex-start"
            alignItems="stretch"
          >
            {body}
          </Grid>
        </Grid>
      </Grid>
    </>
  );
}

Content.propTypes = {
  subHeader: PropTypes.any,
  body: PropTypes.any,
  containerClass: PropTypes.string.isRequired,
  headerClass: PropTypes.string.isRequired,
  screen: PropTypes.object,
};

export default Content;
