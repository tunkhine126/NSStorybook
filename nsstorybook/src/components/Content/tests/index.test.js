/**
 *
 * Tests for Content
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import Content from '../Loadable';

const FakeComponent = () => <span>Test Component</span>;

describe('<Content />', () => {
  const screen = {
    header: {
      xs: 5,
    },
  };

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <Content
        subHeader={<FakeComponent />}
        body={<FakeComponent />}
        containerClass="container"
        headerClass="header"
        screen={screen}
      />
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Content
        subHeader={<FakeComponent />}
        body={<FakeComponent />}
        containerClass="container"
        headerClass="header"
        screen={screen}
      />
    );
    expect(firstChild).toMatchSnapshot();
  });
});
