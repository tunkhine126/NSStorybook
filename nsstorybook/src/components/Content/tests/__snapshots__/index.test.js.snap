// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> Should render and match the snapshot 1`] = `
<div
  class="MuiGrid-root container MuiGrid-container MuiGrid-align-items-xs-flex-start"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
  >
    <p
      class="MuiTypography-root header MuiTypography-body1 MuiTypography-gutterBottom"
    >
      <span>
        Test Component
      </span>
    </p>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
    >
      <span>
        Test Component
      </span>
    </div>
  </div>
</div>
`;
